Services/source/adcs/adcs_service.c

```diff
@@ -0,0 +1,1112 @@
+/*
+ * adcs_service.c
+ *
+ *  Created on: Jul 08, 2021
+ *      Author: tttru
+ */
+
+#include "adcs/adcs_service.h"
+
+SAT_returnState adcs_service_app(csp_packet_t *packet) {
+    uint8_t ser_subtype = (uint8_t)packet->data[SUBSERVICE_BYTE];
+    int8_t status;
+    SAT_returnState return_state = SATR_OK; // temporarily OK
+
+    ADCS_node_identification* node_id = (ADCS_node_identification*)pvPortMalloc(sizeof(ADCS_node_identification));
```

Comment:
I've got ridden of these mallocs and it seems that adcs use them way more than communications services, so you might easily run to memory problems. I recommend using scopes in the cases. You probably won't need mallocs either.

Reply:
I'm thinking of moving each malloc into its subservice so we dont initialize it from the beginning. We only allocate memory for that subservice when we need it. What do you think?

Reply:
That's what I meant. I don't think you would need malloc either if you do that.
Btw, you wouldn't know about the memory issues unless your code actually steps into the service and subservices. So, even if the code compiles and everything, it might not actually work when a GS command is sent to enter these lines.

Reply:
If I dont use malloc, I dont need those structs either. The point of having those structs is to make it easier to transfer to `packet->data[OUT_DATA_BYTE]`, isn't it? Otherwise I would have to send each data one by one.

Reply:
Nope. See an [example](https://github.com/AlbertaSat/ex2_services/blob/master/Services/source/communication/communication_service.c#L196)

Reply:
Ah I see what you meant. Thank you